shorebird_code_push/lib/src/shorebird_updater_io.dart

```diff
@@ -0,0 +1,177 @@
+import 'dart:async';
+import 'dart:ffi';
+import 'dart:isolate';
+
+import 'package:ffi/ffi.dart';
+import 'package:meta/meta.dart';
+import 'package:shorebird_code_push/src/generated/updater_bindings.g.dart';
+import 'package:shorebird_code_push/src/shorebird_updater.dart';
+import 'package:shorebird_code_push/src/updater.dart';
+
+@visibleForTesting
+
+/// Type definition for [Isolate.run].
+typedef IsolateRun = Future<R> Function<R>(
+  FutureOr<R> Function(), {
+  String? debugName,
+});
+
+/// {@template shorebird_updater_io}
+/// The Shorebird IO Updater.
+/// {@endtemplate}
+class ShorebirdUpdaterImpl implements ShorebirdUpdater {
+  /// {@macro shorebird_updater_io}
+  ShorebirdUpdaterImpl(this._updater, {IsolateRun? run})
+      : _run = run ?? Isolate.run {
+    try {
+      // If the Shorebird Engine is not available, this will throw an exception.
+      _updater.currentPatchNumber();
```

Comment:
This isn't being run in an isolate, is it?  Could this hang their app?

Reply:
I carried this over from the current implementation: https://github.com/shorebirdtech/updater/blob/8e7ec5a9b61a2e03b029f7ec7d245e75960b4ebf/shorebird_code_push/lib/src/shorebird_code_push_io.dart#L25

Reply:
Yeah, so I'm pretty sure that can hang your app, since this construction is done on the main isolate, and any call into the bindings might try to take the config lock and thus block on some other FlutterEngine that is currently running an update with shorebird?  @bryanoltman.

I might just leave a fixme that we may need to wrap that in an isolate call (and thus make construction of the Shorebird engine be a future?)

Reply:
https://github.com/shorebirdtech/updater/blob/main/library/src/updater.rs#L519
https://github.com/shorebirdtech/updater/blob/main/library/src/updater.rs#L143

I think we're pretty good about not holding the config across network requests.  But it looks like in at least one case we do, so this could hang:
https://github.com/shorebirdtech/updater/blob/main/library/src/updater.rs#L543
if the background engine is reporting a success or failure.

If we move all of our networking into a queue and then probably add an ASSERT that we aren't ever holding the config lock while doing networking then calls like this are probably fine.

Reply:
Added the fixme 👍